.github/workflows/release.yml

```diff
@@ -37,43 +41,121 @@ jobs:
           python -m build --sdist
           twine check --strict dist/*
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
+          name: shapely-sdist
           path: ./dist/*.tar.gz
-          retention-days: 30
+          retention-days: 5
+          compression-level: 0
+
+  build_wheels_linux:
+    name: Build wheels on Linux with ${{ matrix.baseimage }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: x86_64
+            baseimage: manylinux2014_x86_64
+            basetag: 2024.08.12-1
+            cibw_build: "*manylinux_x86_64"
+          - arch: x86_64
+            baseimage: musllinux_1_1_x86_64
+            basetag: 2024.08.12-1
+            cibw_build: "*musllinux_x86_64"
+          - arch: aarch64
+            baseimage: manylinux2014_aarch64
+            basetag: 2024.08.12-1
+            cibw_build: "*manylinux_aarch64"
+          - arch: aarch64
+            baseimage: musllinux_1_2_aarch64
+            basetag: 2024.08.12-1
+            cibw_build: "*musllinux_aarch64"
 
-  build_wheels:
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: arm64
+        if: ${{ matrix.arch == 'aarch64' }}
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+          buildkitd-flags: --debug
+
+      - name: Build Docker image with GEOS
+        # using build-push-action (without push) to make use of cache arguments
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ci/Dockerfile
+          platforms: linux/${{ matrix.arch }}
+          tags: ${{ matrix.baseimage }}_geos:${{ env.GEOS_VERSION }}
```

Comment:
Including the GEOS_VERSION here is a way to ensure it actually gets rebuilt when we update that number?

Reply:
It's perhaps a legacy of how I was handling this in #2089. It can be `latest` as it's a local image (same as pyogrio).

Reply:
Yes, but I remember from pyogrio that sometimes I just need to do a dummy edit to the dockerfile (or update the base image's tag) just to ensure we rebuild the docker image to get an updated GDAL version. SO it _might_ be that including the geos version here would help with that (i.e. to avoid it still uses the cached image when updating the GEOS version, as that would then not actively test it with the update version)

Reply:
ok, keep this tag for now. I'm still exploring the caching for these actions, as the qemu matrix items take over an hour to build the images.